### ISSUE TITLE
panic on unexpected rule

### ISSUE THREAD

**Issue by yoavsv** [fdaines/arch-go#1]
Hello,

arch-go is a very nice initiative and I would like to use it :)

I tried the below rule on my code and arch-go panicked:

```yml
dependenciesRules:
  - package: "**.my.package"
    shouldNotDependsOn:
      - "*.(?!my-site).*$.*"
```

output was:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x1110c0b]

goroutine 1 [running]:
regexp.(*Regexp).doExecute(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0001884c1, 0x37, 0x0, 0x0, ...)
        /usr/local/Cellar/go/1.16.3/libexec/src/regexp/exec.go:527 +0x56b
regexp.(*Regexp).doMatch(...)
        /usr/local/Cellar/go/1.16.3/libexec/src/regexp/exec.go:514
regexp.(*Regexp).MatchString(...)
        /usr/local/Cellar/go/1.16.3/libexec/src/regexp/regexp.go:525
github.com/fdaines/arch-go/internal/impl/dependencies.(*DependencyRuleVerification).checkComplianceWithRestrictedInternalImports(0xc0005c0190, 0xc0001884c1, 0x37, 0x0, 0xc0004a0b01, 0xc00049ef01)
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/impl/dependencies/dependency_verification.go:93 +0x3e7
github.com/fdaines/arch-go/internal/impl/dependencies.(*DependencyRuleVerification).Verify(0xc0005c0190, 0xc00013dca0)
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/impl/dependencies/dependency_verification.go:56 +0x2b6
github.com/fdaines/arch-go/internal/impl.CheckArchitecture.func1()
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/impl/rule_runner.go:40 +0x18d
github.com/fdaines/arch-go/internal/utils.ExecuteWithTimer(0xc00013dd70)
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/utils/timer_functions.go:12 +0x50
github.com/fdaines/arch-go/internal/impl.CheckArchitecture(0x0)
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/impl/rule_runner.go:25 +0x4e
github.com/fdaines/arch-go/internal/cmd.runCommand(0x158c980, 0x15bf000, 0x0, 0x0)
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/cmd/root.go:35 +0x25
github.com/spf13/cobra.(*Command).execute(0x158c980, 0xc000012210, 0x0, 0x0, 0x158c980, 0xc000012210)
        /Users/yoavsadeh/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x158c980, 0x10013b0, 0x0, 0x0)
        /Users/yoavsadeh/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/yoavsadeh/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
github.com/fdaines/arch-go/internal/cmd.Execute()
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/internal/cmd/root.go:24 +0x2d
main.main()
        /Users/yoavsadeh/go/pkg/mod/github.com/fdaines/arch-go@v0.5.1/main.go:8 +0x25
```

I would expect arch-go to print a meaningful error (in this case that go could not compile the regex I tried to feed it) and gracefully shutdown.

**Comment by fdaines** [comment]
Nice suggestion @yoavsv 
If you want to contribute with a regex-validation feature, feel free to create a PR, in other case I can try to solve it in a few days.

**Comment by fdaines** [comment]
Solved in [v0.6.0](https://github.com/fdaines/arch-go/releases/tag/v0.6.0)

**Comment by yoavsv** [comment]
Great 💪, thanks for the update.
